src/Command/Push/PushArtifactCommand.php

```diff
@@ -0,0 +1,284 @@
+  }
+
+  protected function sanitize(Closure $output_callback, string $artifact_dir):void {
+    $output_callback('out', 'Finding Drupal core text files');
+    $sanitizeFinder = $this->localMachineHelper->getFinder()
+      ->files()
+      ->name('*.txt')
+      ->notName('LICENSE.txt')
+      ->in("$artifact_dir/docroot/core");
+
+    $output_callback('out', 'Finding VCS directories');
+    $vcsFinder = $this->localMachineHelper->getFinder()
+      ->ignoreDotFiles(FALSE)
+      ->ignoreVCS(FALSE)
+      ->directories()
+      ->in(["$artifact_dir/docroot",
+        "$artifact_dir/vendor",
+      ])
+      ->name('.git');
+    $drush_dir = "$artifact_dir/drush";
+    if (file_exists($drush_dir)) {
+      $vcsFinder->in($drush_dir);
+    }
+    if ($vcsFinder->hasResults()) {
+      $sanitizeFinder->append($vcsFinder);
+    }
+
+    $output_callback('out', 'Finding INSTALL database text files');
+    $dbInstallFinder = $this->localMachineHelper->getFinder()
+      ->files()
+      ->in([$artifact_dir])
+      ->name('/INSTALL\.[a-z]+\.(md|txt)$/');
+    if ($dbInstallFinder->hasResults()) {
+      $sanitizeFinder->append($dbInstallFinder);
+    }
+
+    $output_callback('out', 'Finding other common text files');
+    $filenames = [
+      'AUTHORS',
+      'CHANGELOG',
+      'CONDUCT',
+      'CONTRIBUTING',
+      'INSTALL',
+      'MAINTAINERS',
+      'PATCHES',
+      'TESTING',
+      'UPDATE',
+    ];
+    $textFileFinder = $this->localMachineHelper->getFinder()
+      ->files()
+      ->in(["$artifact_dir/docroot"])
+      ->name('/(' . implode('|', $filenames) . ')\.(md|txt)$/');
+    if ($textFileFinder->hasResults()) {
+      $sanitizeFinder->append($textFileFinder);
+    }
+
+    $output_callback('out', "Removing sanitized files from build");
+    $this->localMachineHelper->getFilesystem()->remove($sanitizeFinder);
+  }
+
+  protected function commit(Closure $output_callback, string $artifact_dir, string $commit_hash):void {
+    $output_callback('out', 'Adding and committing changed files');
+    $this->localMachineHelper->execute(['git', 'add', '-A'], $output_callback, $artifact_dir, $this->output->isVerbose());
+    foreach (array_merge($this->vendorDirs($artifact_dir), $this->scaffoldFiles($artifact_dir)) as $file) {
+      // This will fatally error if the file doesn't exist. Suppress error output.
+      $this->logger->debug("Forcibly adding $file");
+      $this->localMachineHelper->execute(['git', 'add', '-f', $file], NULL, $artifact_dir, FALSE);
+    }
+    $this->localMachineHelper->execute(['git', 'commit', '-m', "Automated commit by Acquia CLI (source commit: $commit_hash)"], $output_callback, $artifact_dir, $this->output->isVerbose());
+  }
+
+  protected function push(Closure $output_callback, string $artifact_dir):void {
+    $output_callback('out', 'Pushing changes to Acquia Git');
+    $this->localMachineHelper->execute(['git', 'push'], $output_callback, $artifact_dir, $this->output->isVerbose());
```

Comment:
The repository in the current working directory will be left in a strange state after this command executes. There will be a new upstream commit resulting from this command, but the local repository won’t have it. Perhaps we should pull it down? Otherwise, we could at least inform the user.

Reply:
I don't understand. Are you confusing the source and artifact repos? These commands all execute in a dedicated artifact directory (`/tmp/acli-build-artifact`) that the user will never see. We never modify the source repo (i.e. the user's working directory).